






















\documentclass[aps,prl,twocolumn,superscriptaddress]{revtex4}

\usepackage{graphics}
\usepackage{color}
\usepackage{float}















\newcommand{\outlineitem}[1]{}
\newcommand{\bigoutlineitem}[1]{## #1 ##}
\newcommand{\degC}{`^o`C}
 

 
\begin{document}











\title{Irreversible departure from pluripotency can account for much of gene expression heterogeneity in stem cell culture}












\author{Gautham Nair `^a`}




\affiliation{Department of Bioengineering, University of Pennsylvania, Philadelphia, PA 19104- 6321, USA}
\author{Elsa Abranches `^a`}
\author{Ana M. V. Guedes}
\author{Domingos Henrique`^*`}
\affiliation{Instituto de Medicina Molecular and Instituto de Histologia e Biologia do Desenvolvimento, Faculdade de Medicina da Universidade de Lisboa, Lisboa, Portugal}
\affiliation{Champalimaud Neuroscience Programme, Champalimaud Centre for the Unknown, Doca de Pedroucos, Lisboa, Portugal}
\author{Arjun Raj`^\dagger`}
\affiliation{Department of Bioengineering, University of Pennsylvania, Philadelphia, PA 19104- 6321, USA}









\date{\today}

# Abstract #

\begin{abstract}
Mouse embryonic stem (ES) cells display cell-to-cell heterogeneity in expression of some stem cell maintenance transcription factors including *Nanog*, but the nature of cells expressing low levels of these factors remains unclear. Using a *Nanog* reporter mouse ES cell line, we measured the RNA-Seq transcriptional signature correlated to high and low *Nanog* expression (+/-) in cultures containing FGF/Erk inhibitors (2i) and leukemia inhibitory factor (LIF). 

The transcriptome signature of the *Nanog*(-) population indicates a considerable deviation from pluripotency and is not readily compatible with being an intermediate stage towards differentiation.
Using single molecule RNA FISH, we observed that lineage markers are expressed heterogeneously, but intensely, within the *Nanog*(-) subpopulation and we found that around 70\% of *Nanog*(-) cells are negative for *Oct4* mRNA. We suggest that most *Nanog*(-) cells have taken a considerable departure away from the pluripotent ground state and may not be capable of returning, and we propose a new interpretation of existing work on the reversibility of pluripotency factor gene expression heterogeneity. 

`^a` Contributed equally. `^*` \verb+henrique@fm.ul.pt+, `^\dagger` \verb+arjunraj@seas.upenn.edu+
\end{abstract}


\pacs{}




\maketitle



# Introduction #

## Definition of mESCs ##

Mouse embryonic stem (ES) cells are self-renewing, pluripotent cells derived by cell culture techniques from the inner cell mass of mouse blastocysts, the portion of early embryos that gives rise to the embryo proper.

## Heterogeneity in "stemness" ##

It has been argued that the ES cell state is defined by a unique mix of transcription factors [SilvaCell08, HuangDevDyn09].
It therefore came as a surprise that ES cells in culture display cell to cell gene expression heterogeneity in some of the very transcription factors that have come to define them [AriasCSC11]. In live cells, variability as a function of time has been observed for reporters of *Stella* (*Dppa3*), *Rex1* (*Zfp42*) and *Nanog* expression [HayashiCSC08, ToyookaDev08, MarksCell12, ChambersNature07, KalmarPLOSBio09]. The case of *Nanog* is particularly surprising because it is considered part of the very core of the transcriptional network that maintains the stem-like state[JaenischCell08]. 


## Lineage priming ##

Some have suggested that cell-to-cell heterogeneity of transcription factors can allow individual cells in a stem cell culture to explore potential differentiation options in a process known as "lineage priming" [CrossCurrOpin97]. The idea as applied to ES cell culture is that when the expression of an important stem cell transcription factor dips, 
an ES cell is free to transiently adopt an inclination or perhaps some molecular characteristics of more specialized lineages, including activation of cell-fate determination genes, that predisposes it towards lineage commitment.
For example, *Stella*(-) cells are reported to resemble epiblast-derived stem cells [HayashiCSC08], and Chambers et al. provided evidence that *Nanog*(-) cells are indeed predisposed but not necessarily committed to differentiation [ChambersNature07], and Toyooka et al. demonstrated that *Rex1*(-) cells rapidly differentiate into somatic lineages. Kalmar et al. summarized that the biological role of such "priming" could be "to keep a subpopulation of ES cells continuously primed for differentiation without a single cell being precommitted to a particular fate for a long period of time." [KalmarPLOSBio09] However, proving the existence of lineage priming is by its nature fundamentally challenging.

 
## Precarious stem cells ##

An alternative explanation for the more specialized differentiation potential of some cells that spontaneously arise within an ES culture is that these cells have irreversibly departed from pluripotency. If one considers, as Loh and Lim have proposed, that the ES state is a state of active conflict between factors pulling the cell every which way [LohCSC11], then the maintenance of the ES state is reliant on the canceling out of these various forces. Whenever those forces fall sufficiently out of balance, one expects the spontaneous appearance of non-pluripotent cells in the culture. In this view, the observed cell-to-cell heterogeneity is not reversible and has little role in facilitating differentiation and is just a reflection of the ES cell's eagerness to differentiate. 
Counter to this picture, however, are researchers' observations that some cells with seemingly low marker expression revert to the pluripotent state.

## Questions remain ##

Questions therefore remain as to the origin of stem cell marker heterogeneity, what mechanisms could lead to reversible fluctuations when they are observed, and whether these fluctuations can really be thought of as ES cells exploring possible paths of differentiation. Moreover, the therapeutic utility of ES cells requires the ability to precisely control cell fate decisions in cell culture conditions, and thus understanding and taming potential sources of heterogeneity is crucial. 

## Summary of what we did ##

In this paper, we used a ES cell line containing a rapidly-responding fluorescent *Nanog* reporter [AbranchesPlosONE13] to examine the transcriptional profile of *Nanog*-negative ES cells, comparing it to that of *Nanog*-positive ES cells as well as to ES cells subjected to a short 24h differentiation protocol. Our analysis based on flow cytometry, whole genome RNA sequencing (RNA-Seq) and single cell expression profiling via RNA FISH [RajNatMeth08], suggests that most cells that give rise to the *Nanog*-negative transcriptome signature are not transiently "primed" for differentiation but rather irreversibly departed from pluripotency. We also identify a subpopulation amongst the  *Nanog*-negative cells that may be capable of reverting to a fully pluripotent state. Lastly, we critically analyze experimental reports that have attempted to determine whether the process that gives rise to stem cell gene expression heterogeneity is reversible. We suggest that while the experimental data can be interpreted in the context of the reversible lineage priming hypothesis, the data appears more compatible with a much simpler model in which meaningful deviations from pluripotency are mostly irreversible. 


# Results #


# Experiment description #

## Kick-off ##

To assess the extent that marker gene expression heterogeneity corresponds to departure from pluripotency, we sought to determine genome-wide which genes are up-regulated in cells with low levels of *Nanog* expression, and also which pluripotency genes other than *Nanog* have been down-regulated. As a point of reference for examples of changes that might be involved early in differentiation, we also studied cells cultured in differentiation conditions (no LIF) for 24 hours. 


## The Nd line is a faithful reporter of *Nanog* dynamics ##

To separate *Nanog*-positive from *Nanog*-negative cells, we employed a *Nanog* reporter cell line (Nd). Earlier work describes how this line was  derived from the E14tg2a mouse ES cell line by stable integration of a Bacterial Artificial Chromosome containing the genomic region in the vicinity of *Nanog* but with the *Nanog* coding region replaced with a sequence coding for Venus-NLS-PEST (VNP), a rapidly-degrading yellow fluorescent protein analog with a half-life close to that of NANOG itself. (Fig. [[FIG:overview]]A and \citet{AbranchesPlosONE13}). 


## Description of the experiment ##

For the RNA-Seq experiments, we transferred a population of Nd cells from medium supplemented with serum and leukemia inhibitory factor (LIF) to medium supplemented with two Fgf/Erk inhibitors (2i) and LIF. The population, as measured by *Nanog*:VNP fluorescence distribution, reaches a new steady state within a day (Fig. S1A). Two days after the switch to 2i+LIF,  we sorted the cells by VNP fluorescence. We denote the entire population at this point as sample "Stem". The VNP fluorescence histogram shows a large (80-90\%), broad peak of cells with clear VNP fluorescence, and a small shoulder (10-20\%) with low VNP signals 

(Fig. [[FIG:overview]]B). We sorted dark, bright and very bright populations, (*Nanog*:VNP(-), *Nanog*:VNP(+) and *Nanog*:VNP(++) respectively) and processed them immediately for RNA-Seq. Post-sorting fluorescence histograms are shown in Fig. [[FIG:overview]]B.

In parallel to the culture described above, a sample of ES cells grown in 2i+LIF for 24 hours was induced to differentiate by culturing in medium without the 2i inhibitors or LIF for a period of 24 hours. Flow cytometry of this population, which we denote as "Diff",  shows recognizable *Nanog*:VNP-bright and -dark populations, but the peak of the bright population has moved to lower intensities (Fig. [[FIG:overview]]B) . 



## No batch effects ##

We took two biological replicates and we confirmed by principal component analysis that most of the variation in gene expression between our samples corresponds to biological differences rather than batch effects (Fig. S2).


 


# Genome-wide analysis #




## DESeq hit numbers in both samples ##

Taking a genome-wide perspective, we studied 25313 genes for differential expression (see Fig. [[FIG:hitbasics]]A). The fold changes for a number of important pluripotency and lineage marker genes are shown in Fig. [[FIG:overview]]C. 
Owing to high replicate reproducibility and read depth, nearly half the genes with fold changes of at least  `\pm` 40\% in *Nanog*:VNP(+) vs (-) were called as differential expression hits at a false discovery rate (FDR) of 10\%. Of these 3293 hits, 1993 are more highly expressed in *Nanog*:VNP(-) than *Nanog*:VNP(+) and the balance of 1300 are more highly expressed in *Nanog*:VNP(+). The transcriptome signature of *Nanog*:VNP(++) was very similar to that of *Nanog*:VNP(+) (Fig. S3), and therefore we perform our analysis with the *Nanog*:VNP(+) population.  In our differentiation comparison, we calculated 4233 gene hits differentially expressed between the 2i/LIF population (Stem) and after 24h without 2i or LIF (Diff) with 2235 genes down-regulated and 1998 up-regulated. Due to even further reduced replicate variability, these hits include genes with RNA fold changes as small as `\pm`20\%. 


## Relation to earlier transcriptome signatures ##

The gene expression differences we observed between *Nanog*:VNP(-) and (+) cells correlate closely with those from earlier studies that examined heterogeneity in stem cell marker expression. As shown in Fig. [[FIG:hitbasics]]B the RNA-Seq fold changes between *Nanog*:VNP(+) and *Nanog*:VNP(-) are strongly correlated with those reported for *Rex1*-marked heterogeneity. Of other markers used in heterogeneity studies, both *Stella* and *Pecam1*, like *Rex1*, are down-regulated by approximately the same factor as *Nanog* in *Nanog*:VNP(-) cells (Figs. [[FIG:overview]]C,  S4 and Supplemental Table). Only the transcriptome signature of *Hex1*-based reporters [CanhamPLOSBio10, MorganiCellRep13] appears to be different (Fig. [[FIG:hitbasics]]B), and this may be because *Hex1* is not a pluripotency gene.


## Priming genes are at the extremes of change ##

We noticed that many genes considered to be important makers of the different cell types in the early embryo are strongly correlated to *Nanog*:VNP-marked heterogeneity (Fig. [[FIG:overview]]). This appeared initially to fit with the notion of lineage priming. As shown in Fig. S4, the lineage-associated genes from Fig. [[FIG:overview]]C are found towards the extremes of the foldchange-expression level distribution of our hits, and half of the genes with comparably extreme changes have been annotated for roles in development and pattern specification (Fig. S4). Remarkably, 30 *Hox* genes are up-regulated hits in *Nanog*:VNP(-) cells, with expression fold changes between 4 and 16 (See Fig. S5). 
The expression of lineage-specific markers is therefore one of the strongest features in the transcriptome signature of *Nanog*:VNP(-) cells. 



## DIFF has counterintuitive effects on many commitment and stemness genes ##

In contrast, the transcriptome changes after 24 hours of differentiation do not involve upregulation of a large variety of lineage-specific markers (Fig. [[FIG:overview]]C).  After differentiation, quickly responding pluripotency genes such as *Nanog* and *Klf4* were downregulated, and *Fgf5*, a classic epiblast marker, is strongly upregulated, as were DNA methyl transferases *Dnmt3a*, *Dnmt3b*, and the regulator *Dnmt3l* (Fig. [[FIG:overview]]C and Supplemental Table). The expression levels of *Pou5f1* (*Oct4*) and *Zfp42* (*Rex1*) were at most  slightly affected after just 1 day of differentiation, a pattern that was also observed after 1 day in an earlier published differentiation time course (\citet{HayashiCell11} and Fig. [[FIG:hitbasics]]B for genome-wide correlation). More remarkable is that differentiation for 24 hours leads to down-regulation of many of the lineage markers we found up-regulated in *Nanog*:VNP(-) cells, such as those in Fig. [[FIG:overview]]C. Also, nearly all *Hox* genes are down-regulated (Fig. S5). 


## *Nanog*:VNP(+/-) gene foldchange dynamic range ##

Our analysis of the genes down-regulated in the *Nanog*:VNP(-) population gives a first suggestion that its transcriptome signature corresponds to more than a very brief deviation from pluripotency. The downward fold-changes of these genes in cells negative for  *Nanog* expression are larger than immediately apparent from the *Nanog*:VNP(-) RNA-Seq data because the true fold-changes are obscured by reporter false negatives of the kind that have also been observed for other *Nanog* reporter cell lines [FaddahCSC13]. We found by single molecule RNA FISH that approximately 50\% of *Nanog*:VNP mRNA(-) cells in 2i + LIF are actually *Nanog* mRNA(+) and have the same distribution of *Nanog* RNA levels as the *Nanog*:VNP mRNA(+) population (Fig. S6). On the other hand, VNP transcripts themselves are at least ten times less abundant in *Nanog*:VNP(-) than in *Nanog*:VNP(+), which establishes both the purity of our VNP sort and the close correlation between VNP protein and RNA levels. The 50\% of cells that are *Nanog*:VNP mRNA(-) but *Nanog* mRNA(+) should nominally have a similar gene expression pattern as *Nanog*:VNP mRNA(+) cells (which we found were always *Nanog* (+)) since they differ only in expression of the reporter but not of the gene being reported on. Therefore, even a gene that loses all expression in *Nanog* negative cells will show only a reduction to about 50\% in our RNA-Seq comparison. Remarkably, the downward fold-changes of many pluripotency genes are up against this limit (Fig. [[FIG:overview]]C), suggesting that their transcripts are nearly depleted in true *Nanog* expression negative cells. The requirement for sufficient time to achieve such appreciable decay of transcript numbers suggests that most excursions from *Nanog* positive to *Nanog* negative states are not brief.







## GO analysis ##

To compare the makeup of the sets of hit genes in both our *Nanog*:VNP(+)/*Nanog*:VNP(-) and Stem/Diff comparisons in more general terms, we developed a procedure to uniquely assign hits to a computationally identified handful of Gene Ontology (GO) categories (see Methods). At least 220 of the 1993 genes up-regulated in the*Nanog*:VNP(-) population fell into the "embryo development" GO category, and of those that did not, almost 150 are annotated for cell adhesion (Fig. [[FIG:hitbasics]]C).
For comparison, only slightly more than 100 of the 1998 genes up-regulated after 24h of differentiation are annotated for embryo development. 
Conventional null hypothesis testing finds that the GO terms with the smallest p values associated with gene up-regulation in *Nanog*:VNP(-) and Diff are terms for anatomical development and morphogenesis.


## transcription factor ChIP binding  ##

The large number of hits in our screen led us to ask how many could possibly be direct targets of the pluripotency transcriptional network. To explore this possibility, we first found which genes are bound by pluripotency factors in mouse ES cells using published data from \citet{KimCell08}.We then calculated what percentage of these genes were differential expression hits (Fig. [[FIG:hitbasics]]D).

We found that pluripotency transcription factor binding was not predictive of whether a gene is up-regulated in *Nanog*:VNP(-), but notably increases the chances that the gene is a hit for up-regulation in *Nanog*:VNP(+) or for up- or downregulation after one day of differentiation (Fig. [[FIG:hitbasics]]D). *Nanog*:VNP(-) hit genes are therefore as a set not enriched for binding of the pluripotency transcription factors. Although some early lineage-specification genes, like the ones shown in Fig. [[FIG:overview]]C, are highly up-regulated in *Nanog*:VNP(-) and are bound by pluripotency transcription factors, the fact that such a large fraction of *Nanog*:VNP(-) hits are not in direct contact with the pluripotency network comes as a surprise.  
The pluripotency transcription factors are known to bind a large number of genes that are repressed in the pluripotent state and up-regulated upon differentiation [KimCell08, LohCSC11], and, accordingly, genes bound by these factors are clearly enriched amongst the genes we found up-regulated after 1-day of differentiation. In the context of the reversible lineage-priming hypothesis, *Nanog*:VNP(-) should resemble a state of very early or tentative differentiation and is therefore expected to include many pluripotency transcription factor-repressed genes. One possibility is that these *Nanog*:VNP(-) hit genes as a set follow a different regulation paradigm not directly involving the pluripotency factors but are nevertheless involved in only the early steps of differentiation or lineage priming. What we consider a more likely explanation is that the *Nanog*:VNP(-) transcriptome signature represents an advanced departure from pluripotency to the point that most of its up-regulated genes are beyond the regulatory stage at which the pluripotency network is active. For example, as normal development proceeds, transcriptional regulation is handed off from the pluripotency network to second, third and later wave lineage-specific gene expression cascades which can activate genes that need not have been repressed by the pluripotency network. This interpretation fits cohesively with evidence we present below based on single cell measurements.




## Fluctuation and differentiation have many incongruent genes ##

Our analysis shows a number of clear differences between the transcriptome signatures of 1 day differentiation and cell-to-cell gene expression heterogeneity. Interestingly, one third of the 1352 genes that are 10\% FDR hits in both the *Nanog*:VNP(+/-) and Stem/Diff comparisons actually moved in "opposite" directions( Fig. [[FIG:coherence]]A). These 459 genes are equally split between genes that are up in *Nanog*:VNP(-) but down-regulated after 1 day differentiation in the bottom right-quadrant, including many known lineage-associated genes, and genes that are down in *Nanog*:VNP(-) but up in Diff in the top-left quadrant. The existence of such a large fraction of "incoherent" genes is easy to explain if the spontaneous transition from *Nanog*:VNP(+) to *Nanog*:VNP(-) involves a qualitatively different transcriptional program from that involved in culture-wide differentiation. However, it could also be that these genes are up-regulated and then down-regulated (or viceversa) as differentiation progresses and the "incoherence" could be accounted for by a difference in stage of differentiation between our *Nanog*:VNP(-) and Diff populations. Further work will be necessary to definitively answer this question. 

## lincRNA ##

Our RNA sequencing data also revealed changes in the expression of other transcripts beyond mRNAs, such as long intergenic non-coding RNA (lincRNAs). Studies of lincRNAs expressed in ES cells indicate that they may be important regulators of pluripotency and embryonic differentiation [GuttmanNature11, MaamarGenesDev13]. We analyzed the differential expression of long intergenic non-coding RNA (lincRNA) in both heterogeneity and differentiation. Out of 166 ES-cell specific lincRNA (\citet{GuttmanNature11} and Guttman, personal communication), 46\% are significant hits in one of our comparisons. Surprisingly, 
the ES-cell specific lincRNA are biased towards lower expression in *Nanog*:VNP(-) but higher expression in differentiated cells (Fig. S7).





## Propose a model for what is happening ##


Our analysis so far suggested to us that the transcriptome signature of *Nanog*:VNP(-) cells in 2i+LIF is indicative of cells that have departed from the pluripotent state, rather than representing a transient and shallow departure from pluripotency. First, the fold-reduction of  many pluripotency genes in *Nanog*:VNP(-) are larger than would be expected for short-term fluctuations. As a specific illustration, the very low levels of *Nanog* and VNP RNA determined by RNA FISH in the *Nanog*(-) population (Fig. S6B and S8A) suggest that most *Nanog*(-) cells have stopped transcribing *Nanog* for a time equivalent to several *Nanog* RNA lifetimes. Lastly, the transcription factor binding site analysis above showed many of the genes that are differentially expressed in *Nanog*:VNP(-) cells are not under the proximal control of the pluripotency network, suggesting a relatively advanced departure from pluripotency.




# Single molecule RNA FISH #


Even though we analysed purified subpopulations of ES cells expressing different *Nanog*:VNP levels, we suspected that population averaging as well as the presence of false negative cells could bury evidence for strong deviations from pluripotency beneath the relatively mild RNA-Seq fold changes we found between *Nanog*:VNP(+) and *Nanog*:VNP(-) samples. Regarding pluripotency genes, our observation of *Nanog*:VNP(-)Nanog(+) false negatives within *Nanog*:VNP(-) and the similarity in magnitude of downward RNA-Seq fold-changes of classical pluripotency genes was highly suggestive but not conclusive that the levels of these pluripotency genes, including *Oct4*, are all very low in *Nanog*(-) cells. Regarding lineage-associated genes, we observed upwards fold-changes in *Nanog*:VNP(-) mostly between factors of 4 and 8. Underlying this could be low-level but somewhat uniform expression of these genes in large fractions of *Nanog*:VNP(-) cells, at the level of few transcripts per cell. Instead, we considered it more likely that these lineage-associated gene may be expressed in very small this is not needed, I thinkand possibly exclusive subpopulations within *Nanog*:VNP(-) that represent relatively advanced stages of lineage commitment. We were also very interested to see if lineage-associated gene expression arises mostly in cells that have been thoroughly depleted of pluripotency gene expression. 

To overcome population averaging and make these determinations, we used multiplex single molecule RNA FISH to simultaneously count the number of mRNA of specific subsets of genes in single ES cells from the Nd cell line. To avoid reporter issues, we imaged unsorted populations and compensated by imaging on the order of a thousand cells in each experiment in order to sample infrequently occurring cell expression patterns.



## Oct4 RNA FISH ##

First, we checked whether cells that are negative for *Nanog* had also lost expression of other important pluripotency transcription factors. 

We found that about 70\% of *Nanog*(-) cells in 2i culture were also negative for *Oct4* mRNA (Fig. [[FIG:fish]]A) and virtually all the *Oct4*(-) cells were also *Nanog*(-). Since the *Oct4* mRNA half-life is quite long, with little appreciable decay even in a 6 hour observation window [AbranchesPlosONE13], the lack of *Oct4* mRNA means that these cells stopped transcribing *Oct4* for at least a day or more. Since maintaining *Oct4* gene expression within a tight concentration interval is considered essential to the ES state [NiwaNatGen00], these cells have likely exited pluripotency. We also found that *Nanog*(-) cells also do not express the NANOG target *Rex1* (Fig. S6C), which is consistent with our comparison of RNA-seq data (Fig. [[FIG:hitbasics]]B). and further argues that many if not all *Nanog*(-) cells have undergone more than a transient departure from the pluripotent *Nanog*(+) state. The strong correlations between *Nanog*, *Rex1* and *Oct4* at the single level shows that their switching from (+) to (-) is part of a concerted deviation from pluripotency rather than the effect of standalone transcriptional bursting. 


## Maturity of differentiation programs by RNA FISH ##

To assess the uniformity of lineage marker expression within the *Nanog*(-) subpopulation, we counted *Nanog*, *Crabp2* (ectoderm lineage) and *T* (*Brachyury*, mesendoderm lineage) transcripts in `N=2360` Nd cells grown in 2i+LIF using single molecule RNA FISH. We found that only a few cells within the set of *Nanog*(-) cells express any particular lineage priming gene but when they do, it is at the level of tens (*Crabp2*) or hundreds (*T*) of transcripts (See Fig. [[FIG:fish]]B and S8A). The majority of cells expressing *T* and the vast majority expressing *Crabp2* are *Nanog*(-), implying that the lineage-specific gene expression observed in our *Nanog*:VNP(-) RNA-seq data occurs primarily in small subsets within the *Nanog*:VNP(-)*Nanog*(-) subpopulation. Interestingly, we observed 2/2360 cells that simultaneously expressed both of these lineage markers.


We further investigated the maturity of the lineage specification programs that are active in *Nanog*(-) cells by looking for coexpression of genes belonging to overlapping lineages using the specific example of the mesodermal markers *T* and *Tbx6* [ChapmanDevBio96]. Out of 764 cells, only two were *Tbx6*(+) and they were a subset of the seven *T*(+) cells (Fig. [[FIG:fish]]C). We note that *T* and *Tbx6* are independent mesoderm markers in the sense that the onset of *Tbx6* expression does not require the activity of *T* [ChapmanDevBio96], suggesting that a full-scale mesodermal program is underway in these few cells. 


Especially considering that these cells are also *Oct4*(-), we conclude that the *Tbx6* up-regulation in *Nanog*:VNP(-) cannot be interpreted as lineage priming that will spontaneously revert back to the pluripotent "unprimed" ground state. This may also be the case for many more genes up-regulated in *Nanog*:VNP(-) cells.


# Discussion #

 In this paper, we have studied whether the transcriptional signature of *Nanog*:VNP(-) cells corresponds to the transiently "primed" states. Lineage priming is the notion that some cells in culture transiently deviate from a purely pluripotent expression profile and are "primed" to react to differentiation cues in the environment but will spontaneously revert if such a cue is not received.
We argue that most of the transcriptional signature associated with *Nanog*:VNP(-) ES cells grown in 2i+LIF occurs in a sub-population of cells that has irreversibly transitioned out of pluripotency, rather than being primed in the reversible sense. These partly differentiated cells may best be avoided when trying to direct differentiation in particular stem cell applications.



We do not however rule out the possibility that some reversible lineage priming may occur even within the cultures we studied. In fact, we identified that 70\% of our *Nanog*(-) cells are *Oct4*(-), so conversely 30\% are *Oct4*(+), and conceivably capable of returning to a pluripotent state. However, future work will be needed to establish whether reversible lineage priming occurs in this subpopulation.

The difficulty in proving the existence of "lineage priming" is that experimental limitations force assays for lineage inclination and spontaneous reversion to be carried out separately, but the theory requires that these two phenomena can be manifest by the same cell. In earlier reports, researchers have shown that some cells negative for a pluripotency gene reporter have clear features of inclination to certain lineages, as judged by transcriptome changes and differentiation assays. Researchers have also shown that some cells can spontaneously lose and regain expression of that pluripotency gene reporter. However, these two phenomena could arise from mutually exclusive  subpopulations, a situation that has been found to occur in other developmental contexts [PinaNatCellBio12]. 

Time course data strongly suggests that a considerable fraction of cells in ES culture that become negative for a pluripotency gene reporter will not spontaneously revert back, and it could be that only these non-reverting cells show lineage inclination. Currently, the primary argument that such cells do revert to full pluripotency comes from experiments in which researchers use flow cytometry to sort out the negative subpopulation by the fluorescent reporter, regrow the cells, and show that these cells eventually produce some cells that express high levels of the fluorescent reporter. However, in every instance [KalmarPLOSBio09, ToyookaDev08, HayashiCSC08], including our own (Fig. S1B), the subpopulation that is positive for the pluripotency marker quickly regenerates the entire population, but the low subpopulation does so considerably more slowly (Fig. S10). We derive in the supplemental discussion that in this scenario the rate of return from the pluripotency negative states is surprisingly much slower than even the slow rate of population recovery might lead one to believe. The population recovery curves grossly exaggerate the rate of switching back to a pluripotent ground state. For a typical case in which the marker-positive population regenerates the steady-state on a time-scale of a day but the marker-negative population only does so on the timescale of 2, 3 or more days, the calculated average time of switching of a negative cell to positive is 10 days or longer, not 2 or 3 (see Supplemental Discussion and Fig. S11). The slow population recovery rates usually observed from purified negative populations therefore imply extremely slow rate constants at the cell level. Another explanation for the differing rates of population recovery is that in reality there are subpopulations of marker negative cells, some that can spontaneously revert to pluripotency and others that are irreversibly departed.  Our timelapse results (to be reported elsewhere) are consistent with this interpretation. Some of our *Nanog*:VNP(-) cells rapidly fluctuate back to *Nanog*:VNP(+), whereas some remain stuck at low levels of *Nanog*:VNP(-) for the duration of our imaging. 



We propose the following model as a simple explanation for most observed phenomena related to cell-to-cell gene expression heterogeneity in ES cell culture. First, some ES cells spontaneously and irreversibly exit pluripotency. That an ES cell culture sheds irreversibly non-pluripotent cells is a commonly acknowledged though as-yet unexplained experimental fact [SmithCSC13]. Such cells will not take over a continuously passaged culture as long as their growth rate is sufficiently slower than that of pluripotent ES cells (See mathematical derivation in Supplemental Discussion), an entirely plausible scenario since ES cells grow so quickly. The non-pluripotent cells will lose or have already lost expression of pluripotency genes and any pluripotency gene reporter construct. Though their continued existence in 2i+LIF culture medium will likely prevent these cells from adopting fates that closely correspond to differentiated cell types in the embryo, some of these cells nevertheless become positive for expression of genes normally involved in lineage commitment, including even fairly advanced stages of differentiation.
Separately, some ES cells in the culture transiently lose and regain expression of an observed pluripotency marker without ever changing their lineage differentiation potential. This could be because of a fluctuation in the expression of the underlying endogenous pluripotency gene that is buffered by redundancy in the gene network, or could be explained in some cases by false negatives of reporter constructs. As we described above for our *Nanog* reporter and \citet{FaddahCSC13} reported for many reporter constructs used in literature to date, many cells that are negative for fluorescence from a pluripotency gene reporter are often still positive for the endogenous pluripotency gene. This model appears consistent with all previous data, including population sorting data, and with the data we have presented here on the transcriptome of the *Nanog*(-) subpopulation. We believe this picture offers a simpler explanation than the hypothesis of reversible lineage priming.

## Does *Nanog* cause the fluctuations? ##

Future work may address the mechanism of spontaneous departure from pluripotency that we and others have observed. At the moment, it is not known whether any particular transcription factor on its own controls the transition out of the *Nanog*(+) state and gives rise to the cell-to-cell gene expression variability we observe. The sheer number of genes that we found down-regulated in the *Nanog*:VNP(-) subpopulation suggests that finding such a master regulator would be a challenge even if it exists. Since *Nanog* is considered greatly important to the pluripotent state, it is conceivable that NANOG protein levels themselves could orchestrate this transition. That hypothesis predicts a strong correlation between the *Nanog*:VNP(+)/*Nanog*:VNP(-) transcriptome signature we observed and transcriptome changes following *Nanog* knockdown. Instead, we found little correlation to the  *Nanog* knockdown gene expression microarray time series reported by \citet{MacArthurNatGen12}, as shown in Fig. S9. Although the lack of correlation could be due to differences in cell line and other experimental details, the existing data is more consistent with a role for *Nanog* as a marker of heterogeneity rather than its principal driver. 

## Send-off ##

It has been a puzzle to explain how fluctuations arise in ES cell culture that are both reversible and simultaneously consequential for differentiation, especially since the in-vivo relevance of these fluctuations is questionable [SmithCSC13]. We propose, based on our data and a reexamination of existing reports, that reversibility to the ground state and substantive lineage-inclination may never occur in the same cells.


# Experimental Procedures #

## Cell culture ##
In this study, we used the Nd ES cell line, a BAC-transgenic line for VNP-tagged *Nanog* gene derived from E14tg2a ES cells [AbranchesPlosONE13]. E14tg2a ESCs (a kind gift from Austin Smith's lab, University of Cambridge, UK) were also used as a negative control for VNP expression. ESCs were routinely expanded in serum+LIF media (GMEM medium (Invitrogen) supplemented with ES-qualified serum (Invitrogen) and LIF), and were transferred to 2i+LIF medium (iStem medium (Stem Cells Inc.) supplemented with LIF) for 48h prior to FACS-sorting and RNA collection. Additionally, ES cells were grown in 2i+LIF medium for 24h, followed by removal of LIF and inhibitors (N2B27 medium) for another 24h (Diff). 

## Flow cytometry and sorting ##

Live cells flow cytometry analysis and sorting experiments were performed as described previously [AbranchesPlosONE13], respectively on a FACS Calibur cytometer (Becton Dickinson) or on a FACS Aria cell sorter (Becton Dickinson). For sorting, VNP low (VNP-), VNP intermediate (VNP+) and VNP high (VNP++) Nd ES cells populations were collected and processed for RNA extraction. Bulk populations of cells grown in 2i/LIF for 48h (All) or in "Diff" conditions were also collected from the sorter without gating for VNP levels and analysed. The whole process was repeated once to obtain a biological replicate. 

## RNA extraction and sequencing ##
Total RNA was extracted from `10^6` cells using a High Pure RNA Isolation kit (Roche Diagnostics) and DNAseI. We prepared libraries for RNA-sequencing by using the Illumina TruSeq kit, which includes poly-adenylation selection, following the manufacturers recommendation. We sequenced the libraries on an Illumina HiSeq 2000. Each sample yielded between 90 and 220 million 100 base paired-end reads.


## Differential Expression ##

Reads were aligned to the mouse genome (mm9 assembly) and transcriptome (obtained from the Refseq, UCSC known gene and VEGA annotations) using the RUM RNA-Seq alignment pipeline with default parameters [GrantBioInf11], and we found 82\% to 84\% of reads from each sample mapped uniquely. We used DESeq (v. 1.10.1) to test for differential expression in our RNA-Seq study [AndersGenomeBio10]. 
We selected only one transcript model for each gene, preferring Refseq to UCSC and UCSC over VEGA, and then selecting the longest transcript. We used default DESeq size factor estimation, and estimated count variances with the "per-condition" method, "parametric" fit type, and conservatively choosing the "maximum" sharing mode. We chose not to exploit the paired structure of our experimental design to gain further statistical power. Genes were tested for differential expression between pairs of the five conditions (*Nanog*:VNP(-), *Nanog*:VNP(+), *Nanog*:VNP(++), Stem, and Diff) to produce log fold expression changes and p-values.  Differential expression hits were obtained by controlling for the false discovery rate by the Benjamini-Hochberg procedure.

## Principal Component Analysis ##

Principal Component Analysis was carried out as in the DESeq vignette after re-estimating variances using the required "blind" method and applying the DESeq variance stabilizing transform. 

## GO Analysis ##

For our "greedy" style analysis, we created a ranked list of GO categories by selecting first GO categories larger (annotated for more genes) than some size (for example 750 genes in Fig. [[FIG:hitbasics]]), and then sequentially selecting the GO category with the highest concentration of genes in a gene subset of interest (for example, all genes that were found as hits in either *Nanog*:VNP(+) vs *Nanog*:VNP(-) or Stem vs Diff). After each step we remove all genes in the selected category and repeat until the biological process root category (GO:0008150) is selected. This procedure prevents closely related GO terms from crowding the list of GO terms. Then, given any gene, we assign it to exactly one of these selected categories, preferring the category that was selected earliest. We say that the gene "fell" into that category.

  We also carried out more conventional null hypothesis testing for significant enrichment of GO (Biological Process) categories in gene subsets with the R package goseq [YoungGenomeBio10], using the logarithm of the average read counts of each gene in the conditions considered to account for selection bias. GO annotations were obtained from the Bioconductor package org.Mm.eg.db (v.2.7.1).

## lincRNA ##
Recent versions of lincRNA annotations were obtained by personal communication with Mitch Guttman and Pamela Russell. The transcripts represent updated versions of those published for mouse ES cells [GuttmanNature11]. For each lincRNA, we used a "merged" transcript model constructed from the genomic union of its isoforms. To obtain fold changes and differential expression p-values, the entire DESeq procedure was repeated after adding these merged lincRNA and their counts to the existing gene set. 

## RNA FISH ##
RNA FISH was carried out largely as reported previously [RajNatMeth08]. Cells were released from dishes by trypsinization, washed in PBS, fixed in 4\% paraformaldehyde at room temperature and permeabilized and stored in 70\% ethanol at 4\degC. All washes and hybridizations were carried out in suspension. Wash buffers included 0.1\% Trixon X-100 to minimize losses to sticking on the walls. Samples were mounted between coverglasses in the glucose-oxidase-based 2xSSC anti-fade buffer described previously [RajNatMeth08]. We imaged using a 100x 1.4NA oil-immersion objective, a Nikon Ti-E wide field microscope, and a deep-depletion CCD camera (Pixis 1024, Princeton Instruments ).



# Acknowledgments #

\begin{acknowledgments}
AR acknowledges the support of an NIH New Innovator Award 1DP2OD008514-01 and a Burroughs-Wellcome Fund Career Award at the Scientific Interface.
GN was supported by as a Howard Hughes Medical Institute Postdoctoral Fellow of the Life Sciences Research Foundation. This work was supported by Funda\c{c}\~ao para a Ci\^encia e Tecnologia, Portugal  (SFRH/ BPD/78313/2011 to EA, SFRH/BD/80191/2011 to AMG and PTDC/SAU/OBD/100664/2008). We also thank Hyun Youk and Chris Hsiung for insightful comments on the manuscript.
\end{acknowledgments}


\bibliography{Nanog}

# Figures #

\pagebreak

\begin{figure}[H]

\caption{\label{FIG:overview} \textbf{Experiment Diagram and Differential Expression of Selected Genes} (A) Simplified diagram of the construction of the Nd *Nanog* reporter cell line [AbranchesPlosONE13]. In the Nd cell line Venus (VNP) fluorescence is a reporter of *Nanog* expression. (B) Diagram of the samples from which RNA was extracted for RNA-Seq and flow cytometry of Venus protein fluorescence.  The solid grey profile in all panels corresponds to fluorescence cytometry of the parent E14tg2a cell line as a background. In the Diff panel, the Stem data is shown with a grey outline for comparison.  (C) RNA expression fold changes between the *Nanog*:VNP(+) and *Nanog*:VNP(-) samples and between the Stem and Diff samples for several genes involved in pluripotency (or inner cell mass) or in early extraembronic and embryonic lineages. Dots mark genes that also turn up as hits for significant differential expression at a 10\% false discovery rate}
\end{figure}


\begin{figure}[H]

\caption{\label{FIG:hitbasics} \textbf{Genome Wide Analysis of Heterogeneity and Differentiation}
 (A) Distribution of fold changes for all genes in our study between *Nanog*:VNP(+) and *Nanog*:VNP(-) subpopulations (left) and between Stem and Diff conditions (right). Black regions indicate genes that are significant hits at a 10\% false discovery rate.  

(B) Joint distribution of fold changes for genes between this work and literature data. Left 2 panels include only genes that are hits in *Nanog*:VNP(+)/*Nanog*:VNP(-) and compare *Nanog*:VNP(+)/*Nanog*:VNP(-) fold change to reported RNA-Seq fold changes for *Rex1*+/*Rex1*- populations grown in serum+LIF [MarksCell12] and calculated RNA-Seq fold changes between HV+/HV- subpopulations (HV=*Hex*-Venus reporter) grown in 2i+LIF  [CanhamPLOSBio10]. Right panel: Includes only genes that are hits in Stem/Diff and compares to reported microarray fold changes between ESCs and day1 EpiLCs [HayashiCell11].

(C) The set of GO categories selected by our "greedy" method for genes that are hits in either *Nanog*:VNP(+)/*Nanog*:VNP(-) or Stem/Diff, using a minimal GO size of 750 genes, in the order that they were selected (most relevant first). The biological process parent category occurs last and is omitted from the plot. Leftmost panel: Fraction of genes in GO category that are a hit, with the vertical line showing the (background) fraction of all genes that are hits. Right 4 panels: Within each panel, each hit gene is assigned to the topmost category that it is annotated for.

(D) For each transcription factor, number of genes called as bound in literature [KimCell08] that are also hits of the specified type, divided by the number of bound genes. Black vertical lines indicate the fraction of hits in our gene universe.
}
\end{figure} 

\begin{figure}[H]

\caption{\label{FIG:coherence} \textbf{Distribution of Joint Hits and Differential Expression of lincRNA}
(A) Joint distribution of fold changes for genes that turned up as 10\% FDR hits for differential expression in heterogeneity (*Nanog*:VNP(+) vs *Nanog*:VNP(-)) and differentiation (Stem vs Diff). Text annotations for each quadrant note some genes found in it, as well as the top 3 GO terms and number of genes falling in them by the "greedy" method.

(B) Differential expression analysis for lincRNA identified by \citet{GuttmanNature11} as repressors of certain lineage programs in mouse ES-cells.
}
\end{figure}


\begin{figure}[H]

\caption{\label{FIG:fish} \textbf{Single Cell Analysis of Transcriptome Heterogeneity}
(A) Maximum projection images from Nd cells stained for *Nanog* and *Oct4* RNA. Scale bar in all panels is 5`\mu m` long. The *Nanog*(-)*Oct4*(-) cell in the center is flanked by *Nanog*(+)*Oct4*(+) cells. 

(bottom) Summary of RNA FISH results on 1181 single cells. The vast majority of cells are *Nanog*(+)*Oct4*(+). `16+44=60` cells are *Nanog*(-), and 44 of these are *Oct4*(-). The (+/-) cutoffs were 30 and 80 transcripts for *Nanog* and *Oct4* respectively.

(B) RNA FISH staining for *Nanog* and *Crabp2* (ectoderm marker) showing a *Nanog*(+)*Crabp2*(-) and a *Nanog*(-)*Crabp2*(+) cell. 

(bottom) Summary of simultaneous staining for *Crabp2*, *T* (*Brachyury*) and *Nanog*. The (+/-) cutoffs were 30 and 50 transcripts for *Crabp2* and *T* respectively (Fig. S8).

(C) RNA FISH staining for *Tbx6*, *T* and *Oct4*, showing a *Tbx6*(+)*T*(+)*Oct4*(-) cell.
(bottom) Summary. The (+/-) cutoff for *Tbx6* was 10 transcripts (Fig. S8).
}
\end{figure}


\pagebreak


\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}


\begin{figure}[H]
\begin{center}

\caption{\label{FIG:reconstitution} \textbf{Population relaxation of VNP(+) fraction} 
Fraction of VNP(+) cells determined by flow cytometry following (A) transfer of cells from serum+LIF to 2i+LIF or (B) plating purified VNP(+) and VNP(-) populations grown in 2i+LIF.
}
\end{center}
\end{figure}




\begin{figure}[H]
\begin{center}

\caption{\label{FIG:pca} \textbf{RNA-Seq Principal Components Analysis}. 
Principal component analysis was performed after variance stabilization using DESeq [AndersGenomeBio10]. 
(A)Weights of the 10 samples in the first two principal components of our RNA-Seq experiment. Colors denote different conditions, and the two symbols are the two biological replicates. 
(B) Relative variance of all principal components. Inset shows the weights of the third principal component using the same symbology as in A, showing that it captures gene expression changes between replicates. Together, the plots show that VNP-correlated heterogeneity and differentiation account for most of the sample-to-sample variation of gene RNA levels. 
}
\end{center}
\end{figure}


\begin{figure}[H]
\begin{center}

\caption{\label{FIG:PLUSS} \textbf{VNP++ and VNP+ are Similar}. 
Histogram for the log fold changes of expression by RNA-Seq for 25313 genes between VNP+ and VNP- samples (left) and between VNP++ and VNP+ samples. Only 6 genes are 10\% FDR hits for differential expression between VNP+ and VNP++, nearly three orders of magnitude less than between VNP+ and VNP-.
}
\end{center}
\end{figure}


\begin{figure}[H]
\begin{center}

\caption{\label{FIG:tophits} 
\textbf{The VNP(-) Transcriptome is Distinguished by Lineage-Associated Genes}
(Left) Every point represents a gene that is a 10\% FDR differentially expressed hit between VNP(+) and VNP(-). The y axis is the log fold change from VNP+ to VNP- and the x-axis is the log RPKM in VNP+ for genes higher in VNP+ and in VNP- for genes higher in VNP-. Selected genes are denoted with large symbols, and have the same color if they are markers for the same cell lineages or tissues. The black line is a guide for the eye that separates the more extreme genes from the rest of the population. 
(Right) Fraction of the hits in each sector of the plot on the left that are annotated for at least one of embryo development (GO:0009790), anatomical structure development (GO:0048856), or pattern specification process (GO:0007389). The most salient hits in VNP- are highly enriched for these categories.
}
\end{center}
\end{figure}


\begin{figure}[H]
\begin{center}

\caption{\label{FIG:hox} \textbf{Differential Expression of *Hox* genes}
Fold changes of *Hox* genes in the VNP(+)/VNP(-) and Stem/Diff comparisons. The *Hox* genes are more highly expressed in VNP(-) than VNP(+), but lower in Diff than in Stem. 
}
\end{center}
\end{figure}

\begin{figure}[H]
\begin{center}

\caption{\label{FIG:fishrexvnp} \textbf{Single Cell Analysis of *Nanog*, *Rex1*, and VNP reporter correlation}
  (A) Maximum projection of images from Nd cells stained for *Nanog*, *Rex1*, and Venus fluorescent protein (VNP). Scale bar is 5 `\mu m`. The left-most cell is an example of a VNP reporter false negative. (B) Joint frequency of RNA counts determined by single molecule RNA-FISH for *Nanog* and Venus (VNP) from a measurement on 1177 Nd ES cells grown in 2i+LIF. Note a fraction of cells with low VNP RNA counts but normal *Nanog* RNA counts. (C) Joint frequency of *Nanog* and *Zfp42* (*Rex1*) RNA from the same set of cells. These RNA levels are highly correlated. 
}
\end{center}
\end{figure}

\begin{figure}[H]
\begin{center}

\caption{\label{FIG:speciallincs} 
\textbf{Differential Expression of certain lincRNA}
(Left) Fold changes of lincRNA genes identified by \citet{GuttmanNature11} as associated with the pluripotency network. (Right) Fold change distributions and hit status for all ES-cell lincRNA (\citet{GuttmanNature11} and Guttman, personal communication).
}
\end{center}
\end{figure}


\begin{figure}[H]
\begin{center}

\caption{\label{FIG:fishThresholds} \textbf{Experimental transcript number distributions of *Nanog*, *T*, *Crabp2*, *Oct4*, and *Tbx6*} 
(A) Transcript number distributions for an experiment in which *Nanog*, *T*, and *Crabp2* were probed simultaneously. The cutoff for *Nanog*(+/-) is shown as a straight vertical line on the plot at the left, and the *T* and *Tbx6* distributions are shown separately for *Nanog*(+) and *Nanog*(-) cells.
(B) Similar to A for an experiment in which *Oct4*, *T*, and *Tbx6* were probed simultaneously.
}
\end{center}
\end{figure}



\begin{figure}[H]
\begin{center}

\caption{\label{FIG:mac} \textbf{Comparison to Nanog Knockdown Time Course}
(top) Joint frequency of fold changes observed by microarray by MacArthur and Sevilla et al. at specified time points after external down-regulation of *Nanog*, against our RNA-Seq fold changes between VNP(+) and VNP(-) for genes that are 10\% FDR hits in the VNP(+)/VNP(-) comparison. For the literature data, we used the average of all replicates and all probes corresponding to each gene. (bottom) Same but showing only 10\% hits and fold changes between Stem and Diff conditions.
}
\end{center}
\end{figure}


\begin{figure}[H]
\begin{center}

\caption{\label{FIG:RecoveryCurveCalc} \textbf{Calculated curves for a model of population recovery with no growth differences}. The recovery from (+) and expected recovery from (-) are calculated assuming that they are part of a completely reversible equilibrium of cells that grow at the same rate. In experiments, the actual recovery from (-) is found much slower than expected given the recovery from (+).
}
\end{center}
\end{figure}


\begin{figure}[H]
\begin{center}

\caption{\label{FIG:GeneralCurves} \textbf{Calculated population recovery curves allowing for growth rate differences.} `\kappa` is the rate constant for recovery from a small perturbation from steady state, and `k_{BA}` is the rate constant for individual (-) cells switching to (+). The left and right panels are calculations for steady state (+) fractions of 0.5 and 0.75 respectively (`\alpha^* = 0.5` and `0.75`). The blue line corresponds to no growth rate difference `k_A = k_B`, and the red lines show the effect of increased growth rate of population `A` with respect to `B`. See supplementary discussion for a more thorough explanation. 
}
\end{center}
\end{figure}



\renewcommand{\thetable}{S\arabic{table}}

\begin{table}[H]
\begin{center}

\caption{\textbf{Raw RNA-Seq counts} Number of reads mapped to each of the genes we studied for all samples.
}
\end{center}
\end{table}


\begin{table}[H]
\begin{center}

\caption{\textbf{processed RNA-Seq data} Transcript annotation (mm9-based) information for all genes studied. Calculated fold changes and p-values between Stem and Diff and between VNP(+) and VNP(-) (high fold changes imply high in Diff or VNP(-)), and false discovery rate cutoffs (adjusted p-values). Also, average RPKM values for all five conditions.
}
\end{center}
\end{table}

\end{document}







